src/app.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * This program has been developed by students from the bachelor
+ * Computer Science at Utrecht University within the Software Project course.
+ *
+ * © Copyright Utrecht University
+ * (Department of Information and Computing Sciences)
+ */
+
+import express, { NextFunction } from "express";
+import { Request, Response } from "express";
+import { Context } from "./context";
+import { installRoute, uninstallRoute } from "./install";
+import { body } from "express-validator";
+import { asyncCatch } from "./utils";
+import { handleValidationResult } from "./validate";
+
+export function buildApp(ctx: Context) {
+  const app = express();
+
+  app.use(express.json());
+
+  app.post(
+    "/install",
+    body("userId").exists().isString(),
+    body("addonId").exists().isString(),
+    handleValidationResult,
+    asyncCatch(installRoute(ctx))
+  );
+
+  app.post(
+    "/uninstall",
+    body("userId").exists().isString(),
+    body("addonId").exists().isString(),
+    handleValidationResult,
+    asyncCatch(uninstallRoute(ctx))
+  );
+
+  app.use(
+    (err: Error, req: Request, res: Response, next: NextFunction): void => {
+      console.error(err);
+      res.status(500).json({ error: err.message });
```

Comment:
Ideally we don't print any message at the 500 page at all, because errors can give away which server technology we use, which is a security risk.

Reply:
> Ideally we don't print any message at the 500 page at all, because errors can give away which server technology we use, which is a security risk.

should be fixed with the last commit now